CMakeLists.txt

```diff
@@ -130,63 +149,77 @@ set(GCC_ARM_NONE_EABI_MD5 6341f11972dac8de185646d0fbd73bfc)
 
 set(JLINK_LINUX_DOWNLOAD_URL ${JLINK_DOWNLOAD_URL}/${JLINK_DEB_FILE})
 
-
 #######################################################################################################################
 # Local names
 #######################################################################################################################
 
-#https://www.nordicsemi.com/-/media/Software-and-other-downloads/SoftDevices/S132/s132nrf52720.zip
-
-#set(SOFTDEVICE_HEX_FILE components/softdevice/s132/hex/s132_nrf52_6.1.1_softdevice.hex)
-#set(SOFTDEVICE_HEX_FILE components/softdevice/s132/hex/s132_nrf52_7.0.1_softdevice.hex)
+# Softdevice is now set in CMakeBuild config file and a different one can be used depending on your target
+# It should be something like:
+#     SOFTDEVICE=s132_nrf52_7.0.1
+#
 set(SOFTDEVICE_HEX_FILE components/softdevice/s132/hex/${SOFTDEVICE}_softdevice.hex)
 
-list(APPEND CMAKE_ARGS "-DSOFTDEVICE_HEX_FILE:STRING=${SOFTDEVICE_HEX_FILE}")
+list(APPEND CMAKE_BLUENET_ARGS "-DSOFTDEVICE_HEX_FILE:STRING=${SOFTDEVICE_HEX_FILE}")
 
 #######################################################################################################################
 # Download dependencies
 #######################################################################################################################
 
-ExternalProject_Add(nordic_sdk
-	DOWNLOAD_DIR ${WORKSPACE_DIR}/downloads
-	SOURCE_DIR ${WORKSPACE_DIR}/tools/nrf5_sdk
-	URL ${NORDIC_NRF5_SDK_DOWNLOAD_URL}
-	URL_MD5 ${NORDIC_NRF5_SDK_MD5}
-	CONFIGURE_COMMAND ${CMAKE_COMMAND} -E echo "Skipping configure step."
-	BUILD_COMMAND ${CMAKE_COMMAND} -E echo "Skipping build step."
-	INSTALL_COMMAND ${CMAKE_COMMAND} -E echo "Skipping install step."
-	PATCH_COMMAND cd ${WORKSPACE_DIR} && patch -p0 < patch/01nrf5.patch
-	COMMAND cd ${WORKSPACE_DIR} && patch -p0 < patch/02nrf5.patch && patch -p0 < patch/03nrf5.patch && patch -p0 < patch/04nrf5.patch
-	)
+if (NORDIC_SDK_VERSION EQUAL 15)
+	ExternalProject_Add(nordic_sdk
+		DOWNLOAD_DIR ${WORKSPACE_DIR}/downloads
+		SOURCE_DIR ${WORKSPACE_DIR}/tools/nrf5_sdk
+		URL ${NORDIC_NRF5_SDK_DOWNLOAD_URL}
+		URL_MD5 ${NORDIC_NRF5_SDK_MD5}
+		CONFIGURE_COMMAND ${CMAKE_COMMAND} -E echo "Skipping configure step."
+		BUILD_COMMAND ${CMAKE_COMMAND} -E echo "Skipping build step."
+		INSTALL_COMMAND ${CMAKE_COMMAND} -E echo "Skipping install step."
+		PATCH_COMMAND cd ${WORKSPACE_DIR} && patch -p0 < patch/01nrf5.patch
+			COMMAND cd ${WORKSPACE_DIR} && patch -p0 < patch/02nrf5.patch
+			COMMAND cd ${WORKSPACE_DIR} && patch -p0 < patch/03nrf5.patch
+		)
+else()
+	ExternalProject_Add(nordic_sdk
+		DOWNLOAD_DIR ${WORKSPACE_DIR}/downloads
+		SOURCE_DIR ${WORKSPACE_DIR}/tools/nrf5_sdk
+		URL ${NORDIC_NRF5_SDK_DOWNLOAD_URL}
+		URL_MD5 ${NORDIC_NRF5_SDK_MD5}
+		CONFIGURE_COMMAND ${CMAKE_COMMAND} -E echo "Skipping configure step."
+		BUILD_COMMAND ${CMAKE_COMMAND} -E echo "Skipping build step."
+		INSTALL_COMMAND ${CMAKE_COMMAND} -E echo "Skipping install step."
+		)
```

Comment:
So we have to create patches for this SDK. Maybe add that to some TODO list for this PR?

Reply:
Good point. Previous patches have not been reapplied. 

I vote for removal of 01nrf5.patch (just use chmod on command line before running the thing) and 02nrf.patch (see reasoning above). 

The other patches, need re-application. Alternatively, the corresponding files can be placed in `src/third/nrf`. In that case no patching is required. I would then recommend `src/third/nrf/sdk15` subdirectories. 

If we keep the patches, we need to create `patch/sdk15` subdirectories because the patches will be different per sdk.

Even if we go over to a new SDK for all our products this makes it possible to still compile older firmware against an older SDK for a while. 

The same is actually true for the `app_config.h` and `sdk_config.h` files. They should also be structured accordingly in `include/third/nrf/sdk15`, etc.

Reply:
Patch 03nrf5.patch is about this issue reported on Nordic forum https://devzone.nordicsemi.com/f/nordic-q-a/20846/getting-service-count-from-database-discovery-module.

See https://github.com/crownstone/nRF5_SDK/blob/17.1.0/components/ble/ble_db_discovery/ble_db_discovery.c and compare with https://github.com/crownstone/nRF5_SDK/blob/15.3.0/components/ble/ble_db_discovery/ble_db_discovery.c to see if it has to be applied again. Or even better, check if BLE_DB_DISCOVERY_AVAILABLE is nowadays properly triggered.

It looks like it should given the new function `discovery_available_evt_trigger`.

Reply:
Problem that is addressed (partly) by `04nrf5.patch` as reported on https://devzone.nordicsemi.com/f/nordic-q-a/71360/softdevice-event-handler-continuously-called-with-nrf_evt_radio_blocked and https://devzone.nordicsemi.com/f/nordic-q-a/80616/scheduler-get-full-by-nrf_evt_radio_blocked-events-during-connect is now on https://github.com/crownstone/nRF5_SDK/tree/17.1.0. There are no other patches in sdk15 that need application in sdk17.